PowerEditor/src/winmain.cpp

```diff
@@ -21,6 +21,78 @@
 #include "verifySignedfile.h"
 #include "NppDarkMode.h"
 
+// Include that contain COM classes and IDLs
+// The COM interfaces that we are about here are only available if we specify Windows 10 or higher,
+// So we override the default _WIN32_WINNT that is set in the project file for this include.
+#if defined(_WIN32_WINNT) && (_WIN32_WINNT < _WIN32_WINNT_WIN10)
+#	undef _WIN32_WINNT
+#endif
+#define _WIN32_WINNT _WIN32_WINNT_WIN10

```

Comment:
Will it work on Windows 7?

Reply:
No idea.

Microsoft only defines the class and interfaces in the header if you specify that you want Windows 10-level support. It's possible that the class and interface are registered on 7, but I don't have a system to check with or to figure out what the class and interface IDs are.

So, on Windows 7, it will either work, or it will do nothing because said class and interface IDs don't exist.

Reply:
Looking further into it, while Windows 7/8.1 do technically support virtual desktops as far as I know, there is no public API for them.

A further implementation would:

1. Require me to have a 7/8.1 system set up.
2. To reverse how to interface with them.

So, I'm not sure how to get the functionality working on them, Microsoft doesn't appear to intend for it.

Unless you meant if this would break the program for Windows 7, in which case, I don't see why it would. While doing that can potentially change some GUIDs, it's being done in a very localized fashion, and only for getting this specific GUID.

Reply:
@ameisen Your PR does not have to bring Virtual Desktop functionality to Windows 7.  Nor should it if extensive coding would be required to do so.  However, your PR must not prevent Notepad++ from running on Windows 7 nor degrade its performance there.

Reply:
Hopefully someone can test it under Windows 7.

Reply:
Well, I added a bit more code to try and make sure that that define gets restored to its original value, and that nothing else is impacted.

I cannot really do any testing, though. It _should_ work, I don't particularly see why it wouldn't, but I cannot assert that it would.

Reply:
Ok. My concern was that the GUID isn't defined on Windows 7. But since we compile it on Windows 10, it doesn't matter.
When running on Windows 7 it seems like `CoCreateInstance` just doesn't succeed and since we're checking the result, everything is fine.

Reply:
Even compiling on Windows 7, even if the header doesn't have the GUID it would just take the fallback route that MinGW uses and define it with the hardcoded value. So, that should _still work_™.